src/include/miopen/solver/ck_utility_common.hpp

```diff
@@ -0,0 +1,145 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2021 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#ifndef GUARD_CK_UTILITY_COMMON_HPP_
+#define GUARD_CK_UTILITY_COMMON_HPP_
+
+#include <miopen/env.hpp>
+#include <miopen/hip_build_utils.hpp>
+#include <miopen/mlo_internal.hpp>
+#include <miopen/rocm_features.hpp>
+#include <miopen/solver/convolution_context_interpreter.hpp>
+#include <algorithm>
+#include <sstream>
+
+#include "../composable_kernel/composable_kernel/include/utility/data_type_enum.hpp"
+#include "../composable_kernel/host/solver/include/convolution_problem_descriptor.hpp"
+#include "../composable_kernel/host/solver/include/solver_common.hpp"
+
+MIOPEN_DECLARE_ENV_VAR(MIOPEN_DEBUG_CK_BLOCK_SYNC_LDS_WITHOUT_SYNC_VMEM)
+MIOPEN_DECLARE_ENV_VAR(MIOPEN_DEBUG_CK_USE_AMD_BUFFER_ADDRESSING)
+
+namespace miopen {
+namespace solver {
+namespace ck_utility {
+
+static inline bool is_ck_supported_hardware(const ConvolutionContext& c)
+{
+    return (StartsWith(c.GetStream().GetDeviceName(), "gfx803") &&
+            c.GetStream().GetMaxComputeUnits() == 64) ||
+           StartsWith(c.GetStream().GetDeviceName(), "gfx900") ||
+           StartsWith(c.GetStream().GetDeviceName(), "gfx906") ||
+           StartsWith(c.GetStream().GetDeviceName(), "gfx908") ||
+           StartsWith(c.GetStream().GetDeviceName(), "gfx90a") ||
+           StartsWith(c.GetStream().GetDeviceName(), "gfx1030");
+}
+
+static inline bool is_support_amd_buffer_atomic_fadd(const std::string& device_name)
+{
+    return StartsWith(device_name, "gfx908");
+}
+
+static inline auto get_ck_common_compiler_flag(const ConvolutionContext& ctx)
```

Comment:
Should we use Handle instead of  ConvolutionContext  so that non-convolution components like Reduction can use them ?

Reply:
done